src/SFA.DAS.ApplyService.Web/Controllers/ApplicationController.cs

```diff
@@ -58,48 +58,44 @@ public async Task<IActionResult> Applications()
 
             if (!applications.Any())
             {
+                //ON-2068 Registered org  with no application created via digital service then
+                //display empty list of application screen
                 if (org != null && org.RoEPAOApproved)
-                {
-                      return await StartApplication(userId);
-                }
+                    return View(applications);
 
-                if (org == null)
-                {
-                    if (await _userService.AssociateOrgFromClaimWithUser())
-                        return await StartApplication(userId);
+                if (org != null)
+                    return View("~/Views/Application/Declaration.cshtml");
 
-                    return RedirectToAction("Index", "OrganisationSearch");
-                }
+                if (await _userService.AssociateOrgFromClaimWithUser())
+                    return await StartApplication(userId);
 
-                return View("~/Views/Application/Declaration.cshtml");
+                return RedirectToAction("Index", "OrganisationSearch");
 
             }
 
+            //ON-2068 If there is an existing application (in progress or approved) for an org that is registered then display it
+            //in a list of application screen
+            if (applications.Count() == 1 && (org != null && org.RoEPAOApproved))
+                return View(applications);
+
             if (applications.Count() > 1)
-            {
                 return View(applications);
-            }
 
             var application = applications.First();
 
-            if (application.ApplicationStatus == ApplicationStatus.FeedbackAdded)
+            switch (application.ApplicationStatus)
             {
-                return View("~/Views/Application/FeedbackIntro.cshtml", application.Id);
-            }
-
-            if (application.ApplicationStatus == ApplicationStatus.Rejected)
-            {
-                return View(applications);
-            }
-
-            if (application.ApplicationStatus == ApplicationStatus.Approved)
-            {
-                return View(applications);
+                case ApplicationStatus.FeedbackAdded:
+                    return View("~/Views/Application/FeedbackIntro.cshtml", application.Id);
+                case ApplicationStatus.Rejected:
+                case ApplicationStatus.Approved:
+                    return View(applications);
+                default:
+                    return RedirectToAction("SequenceSignPost", new {applicationId = application.Id});
```

Comment:
Does this code have test coverage?

Reply:
Probably not, although I can tell that the logic for this part hasn't changed. Making it a switch is tidier